### ISSUE TITLE
bug: initializing customized built-in elements

### ISSUE THREAD

**Issue by dbarnespaychex** [ionic-team/stencil#4089]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.20.0

### Current Behavior

Stencil's [renderer](https://github.com/ionic-team/stencil/blob/main/src/runtime/vdom/vdom-render.ts#L93) doesn't handle [customized built-in elements](https://html.spec.whatwg.org/multipage/custom-elements.html#attr-is) correctly. It should [pass an options argument to createElement](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement#options) specifying the `is` value. React's [parser](https://github.com/facebook/react/blob/main/packages/react-dom-bindings/src/client/ReactDOMComponent.js#L429) handles this special case correctly.

We aren't asking for Stencil to support _creating_ customized built-in elements (see #1382), only to honor existing customized built-in elements.

Example code:
```jsx
return <button is="custom-button">content</button>
```
This should render a button with `custom-button` functionality. Instead, it renders a normal button with an `is` attribute.

### Expected Behavior

Stencil's parser should invoke `createElement` with the options parameter specified when it detects a vDOM node with an "is" attribute.

### Steps to Reproduce

Given someone registering the following customized built-in element:
```ts
class CustomButton extends HTMLButtonElement {
    constructor() {
        super();
    }
    connectedCallback() {
        this.setAttribute('data-custom', '');
    }
}

if (!window.customElements.get('custom-button'))
    window.customElements.define('custom-button', CustomButton, { extends: 'button' });
```
The following code should render a button with a "data-custom" attribute:
```tsx
import { Component, h } from '@stencil/core';

@Component({
    tag: 'my-component',
    styleUrl: 'my-component.css',
    shadow: true,
})
export class MyComponent {
    render() {
        return (<button is="custom-button">content</button>)
    }
}
```
Instead, the Stencil code renders a normal button with an `is="custom-button"` attribute.

### Code Reproduction URL

https://github.com/dbarnespaychex/stencil-bug

### Additional Information

React's parser handles this special case explicitly:
https://github.com/facebook/react/blob/main/packages/react-dom-bindings/src/client/ReactDOMComponent.js#L429

**Comment by rwaskiewicz** [comment]
Thanks @dbarnespaychex!

I've labeled this to be ingested into our backlog and to try to get signal from the community for the general demand for this is. To the best of my knowledge, Stencil never rolled out support for customized built-in elements due to lack of support for [legacy browsers and Safari](https://caniuse.com/custom-elementsv1). With Stencil v3, we've deprecated support for the former, and wouldn't be opposed to revisiting the topic.

**Comment by wederchr** [comment]
@rwaskiewicz I would very much love to see Stencil support this feature. Accessibility is hard and it's even harder with autonomous custom elements. Regarding Safari support: There's an [ongoing discussion](https://github.com/WebKit/standards-positions/issues/97) about this. In the meantime: Fixing https://github.com/ionic-team/stencil/issues/2284 would already be a big help in facilitating accessibility. It is now supported by all major (non-deprecated) browsers.

**Comment by rwaskiewicz** [comment]
Adding the following update - at this time, WebKit has formally announced that they are in opposition of implementing Customized Built-In Elements (https://github.com/WebKit/standards-positions/issues/97) and are currently exploring alternatives. Unless they revert their decision, I do not believe we will be moving forward with supporting them in Stencil. For the time being, I’m going to close this issue to convey our current intent. Should things change, we’d be happy to reopen!

**Comment by dbarnespaychex** [comment]
Hi @rwaskiewicz, thanks for following up.

This issue really encompasses 2 separate but related topics:

- fix the parser to support CBEs
- allow developers to create CBEs

I can understand deciding not to allow developers to _create_ CBEs in Stencil, but why explicitly prevent developers from _using_ CBEs written elsewhere? The Stencil JSX parser should treat the existence of an `is` attribute as a sign to invoke the overloaded version of `document.createElement`. By _not_ doing this, it currently renders invalid DOM. This is a defect against the specification, regardless of what one browser vendor says.

At the risk of digressing, this decision not to support CBE usage in the parser is disturbing because -- coupled with the decision to remove polyfills from builds -- it places Stencil users into a terrible position. If browser support is your new benchmark then will Stencil now wait for the major browser vendors to all implement a web standard before it implements it? What if Firefox takes 2 years to prioritize a feature, or Safari says there's not enough money in it? Why not simply implement the _standard_ and let developers decide if they want to run their components within supporting browsers or apply the necessary polyfills?

To summarize: feel free not to support CBE creation; I disagree but that's a feature request and you can reject it. But do please fix the JSX parser so it invokes `document.createElement` correctly; that is clearly a defect against the spec.

Thank you!
